animata/widget/music-stack-interaction.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react";
+import { Layers, LayoutGrid } from "lucide-react";
+
+const carouselStyles = {
+  perspective: "1000px",
+  overflow: "hidden",
+};
+
+const carouselInnerStyles = {
+  display: "flex",
+  transformStyle: "preserve-3d" as React.CSSProperties["transformStyle"],
+  transition: "transform 1s",
+  justifyContent: "center",
+  alignItems: "center",
+  height: "100%",
+};
+
+const carouselItemStyles = {
+  minWidth: "200px",
+  marginLeft: "-180px",
+  transform: "rotateY(15deg) translateZ(300px) translateX(-50px)",
+  backfaceVisibility: "hidden" as React.CSSProperties["backfaceVisibility"],
+  boxShadow: "0 4px 8px rgba(0, 0, 0, 0.2)",
+  transition: "transform 0.5s, box-shadow 0.5s",
+};
+
+const carouselItemFirstChildStyles = {
+  minWidth: "200px",
+  marginLeft: "0",
+  transform: "rotateY(0deg) translateZ(300px) translateX(0)",
+  backfaceVisibility: "hidden" as React.CSSProperties["backfaceVisibility"],
+  boxShadow: "0 4px 8px rgba(0, 0, 0, 0.2)",
+  transition: "transform 0.5s, box-shadow 0.5s",
+};
+
+const carouselBackgroundStyles = {
```

Comment:
you can add type annotations here instead of adding them on each property:

`const carouselBackgroundStyles: React.CSSProperties`